### PR TITLE
Hometask5 oop movie

### PR DESCRIPTION
**Pull request by AndriiBakayev**

### CODE REVIEW

File: src/oop/app/src/Parsers/KinoukrDomCrawlerParserAdapter.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+     * $title, $poster, $description  u predefined page
+     * class  detects needed content and sets StdObj back to csrapper
+     *
+     * @param string $html content of site
+     *
+     * @return mixed object with following properties: $title, $poster, $description.
+     */
+    public function parseContent(string $html): object
+    {
+        $crawler = new Crawler($html);
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Parsers/KinoukrDomCrawlerParserAdapter.php]: please move Crawler to __construct method as dependency

File: src/oop/app/src/Scrapper.php [hunk 2]
```diff
@@ -1,9 +1,56 @@
...
+    /**
+     * Gets content from url, prrses using Parser class object and sets movie object
+     * @param  string  $url url of the site
+     * @return Movie Movie is an object with title,poster and description of movie
+     */
+    public function getMovie(string $url): Movie
+    {
+        $getHtml = $this->Transporter->getContent($url);
+        $parced = $this->Parser->parseContent($getHtml);
+        $movie = new Movie();
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Scrapper.php]: please move Movie to __constuct method (same as you did with parser)

File: src/oop/app/src/Scrapper.php [hunk 3]
```diff
@@ -1,9 +1,56 @@
...
+        if (is_object($parced)) {
+            $movie->setTitle($parced->title);
+            $movie->setPoster($parced->poster);
+            $movie->setDescription($parced->description);
+        }
+        return $movie;
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Scrapper.php]: please add empty line before return

File: src/oop/app/src/Transporters/CurlStrategy.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        curl_setopt($ch, CURLOPT_URL, $url);
+        curl_setopt($ch, CURLOPT_USERAGENT, $user_agent);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        curl_setopt($ch, CURLOPT_FOLLOWLOCATION, 1);
+        curl_setopt($ch, CURLOPT_HEADER, 0);
+        curl_setopt($ch, CURLOPT_NOBODY, 0);
+        curl_setopt($ch, CURLOPT_TIMEOUT, 30);
+        $res = curl_exec($ch);
+        if ($res === false) {
+                echo 'Error curl: ' . curl_error($ch);
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Transporters/CurlStrategy.php]: to many spaces

File: src/oop/app/src/Transporters/GuzzleAdapter.php [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ */
+class GuzzleAdapter implements TransportInterface
+{
+    /** gets the page content isung Guzzle libulary
+     * @param  string $url for get
+     * @return string Index.html source content
+     */
+    public function getContent(string $url): string
+    {
+        $client = new \GuzzleHttp\Client(['base_uri' => 'http://httpbin.org/','timeout'  => 2.0]);
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Transporters/GuzzleAdapter.php]: please move Guzzle Client into __construct method as dependency

File: src/oop/app/src/Transporters/cacert.pem [hunk 6]
```diff
@@ -0,0 +1,3460 @@
+##
+## Bundle of CA Root Certificates
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Transporters/cacert.pem]: Why did you commit this file ?

File: tests/GetUniqueFirstLettersTest.php [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        //require '/scr/web/functions.php';
+    }
+
+    /**
+     * @dataProvider positiveDataProvider
+     */
+    public function testPositive($input, $expected)
+    {
+       //echo __DIR__ ;
+         require_once __DIR__ . '\..\src\web\functions.php';
```
Comment (yderevianko-cartera) [inline tests/GetUniqueFirstLettersTest.php]: require should not be placed inside method

**Review by yderevianko-cartera — CHANGES_REQUESTED** [review]

File: tests/GetUniqueFirstLettersTest.php (hunk 7 above)
Comment (AndriiBakayev) [reply tests/GetUniqueFirstLettersTest.php]: It's a test from Hometask 6. Deleted.

File: src/oop/app/src/Transporters/GuzzleAdapter.php (hunk 5 above)
Comment (AndriiBakayev) [reply src/oop/app/src/Transporters/GuzzleAdapter.php]: Done

File: src/oop/app/src/Transporters/CurlStrategy.php (hunk 4 above)
Comment (AndriiBakayev) [reply src/oop/app/src/Transporters/CurlStrategy.php]: Done

File: src/oop/app/src/Scrapper.php (hunk 3 above)
Comment (AndriiBakayev) [reply src/oop/app/src/Scrapper.php]: Done

File: src/oop/app/src/Scrapper.php (hunk 2 above)
Comment (AndriiBakayev) [reply src/oop/app/src/Scrapper.php]: Done

File: src/oop/app/src/Parsers/KinoukrDomCrawlerParserAdapter.php (hunk 1 above)
Comment (AndriiBakayev) [reply src/oop/app/src/Parsers/KinoukrDomCrawlerParserAdapter.php]: Done

File: src/oop/app/src/Transporters/cacert.pem (hunk 6 above)
Comment (AndriiBakayev) [reply src/oop/app/src/Transporters/cacert.pem]: It was need to VERIFY_PEER. Switched off and deleted file.

File: src/oop/app/src/Parsers/KinoukrDomCrawlerParserAdapter.php [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+ * @link     https://github.com/AndriiBakayev
+ */
+class KinoukrDomCrawlerParserAdapter implements ParserInterface
+{
+    private Crawler $crawler;
+    private Movie $movie;
+
+    public function __construct(Movie $movie)
+    {
+        $this->crawler = new Crawler();
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Parsers/KinoukrDomCrawlerParserAdapter.php]: please pass Crawler as dependency (do not use new Crawler())

File: src/oop/app/src/Scrapper.php [hunk 9]
```diff
@@ -1,9 +1,53 @@
...
+    /**
+     * Sets transporter and parser objects for future getMogie routine
+     * @param  TransportInterface  $transporter object
+     * @param  ParserInterface  $parser object
+     */
+    public function __construct(TransportInterface $transporter, ParserInterface $parser)
+    {
+        $this->transporter = $transporter;
+        $this->parser = $parser;
+        $this->movie = new Movie();
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Scrapper.php]: please pass Movie as dependency (do not use new Movie())

File: src/oop/app/src/Transporters/GuzzleAdapter.php [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+ * Note: About GuzzleHttp Client you can read here:
+ * https://docs.guzzlephp.org/en/stable/
+ */
+class GuzzleAdapter implements TransportInterface
+{
+    private \GuzzleHttp\Client $client;
+
+    public function __construct()
+    {
+        $this->client = new \GuzzleHttp\Client(['base_uri' => 'http://httpbin.org/','timeout'  => 2.0]);
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Transporters/GuzzleAdapter.php]: please pass Client as dependency (do not use new Client())

**Review by yderevianko-cartera — CHANGES_REQUESTED** [review]

File: src/oop/app/src/Transporters/GuzzleAdapter.php (hunk 10 above)
Comment (AndriiBakayev) [reply src/oop/app/src/Transporters/GuzzleAdapter.php]: Done

File: src/oop/app/src/Scrapper.php (hunk 9 above)
Comment (AndriiBakayev) [reply src/oop/app/src/Scrapper.php]: Done

File: src/oop/app/src/Parsers/KinoukrDomCrawlerParserAdapter.php (hunk 8 above)
Comment (AndriiBakayev) [reply src/oop/app/src/Parsers/KinoukrDomCrawlerParserAdapter.php]: Done

File: src/oop/app/src/ScrapperFactory.php [hunk 11]
```diff
@@ -7,13 +7,23 @@
...
 /** Class which decides which transporter and which parser use for site
  * ScrapperFactory
  */
 class ScrapperFactory
 {
+    private Movie $movie;
+
+    public function __construct()
+    {
+        $this->movieLeft = new Movie();
```
Comment (yderevianko-cartera) [inline src/oop/app/src/ScrapperFactory.php]: why do you need two movies ? just pass new Movie in create method

**Review by yderevianko-cartera — CHANGES_REQUESTED** [review]

File: src/oop/app/src/ScrapperFactory.php (hunk 11 above)
Comment (AndriiBakayev) [reply src/oop/app/src/ScrapperFactory.php]: Done. It was a task not to change index.php

File: src/oop/app/src/ScrapperFactory.php [hunk 12]
```diff
@@ -2,28 +2,50 @@
...
+                return new Scrapper(
+                    new CurlStrategy(),
+                    new FilmixParserStrategy($this->movieLeft),
+                    $this->movieLeft
+                );
             case 'kinoukr':
-                return new Scrapper(new GuzzleAdapter(), new KinoukrDomCrawlerParserAdapter());
+                return new Scrapper(
+                    new GuzzleAdapter(new \GuzzleHttp\Client()),
+                    new KinoukrDomCrawlerParserAdapter($this->movieRight, new Crawler()),
```
Comment (yderevianko-cartera) [inline src/oop/app/src/ScrapperFactory.php]: new KinoukrDomCrawlerParserAdapter(new Movie(), new Crawler()),

File: src/oop/app/src/ScrapperFactory.php [hunk 13]
```diff
@@ -2,28 +2,50 @@
...
+                    $this->movieRight
```
Comment (yderevianko-cartera) [inline src/oop/app/src/ScrapperFactory.php]: new Movie()

File: src/oop/app/src/ScrapperFactory.php [hunk 14]
```diff
@@ -2,28 +2,50 @@
...
 
+/** Class which decides which transporter and which parser use for site
+ * ScrapperFactory
+ */
 class ScrapperFactory
 {
-    /**
+    private Movie $movie;
+
+    public function __construct(Movie $movieLeft, Movie $movieRight)
```
Comment (yderevianko-cartera) [inline src/oop/app/src/ScrapperFactory.php]: not needed method

**Review by yderevianko-cartera — CHANGES_REQUESTED** [review]

File: src/oop/app/src/ScrapperFactory.php (hunk 13 above)
Comment (AndriiBakayev) [reply src/oop/app/src/ScrapperFactory.php]: Done

File: src/oop/app/src/ScrapperFactory.php (hunk 14 above)
Comment (AndriiBakayev) [reply src/oop/app/src/ScrapperFactory.php]: Done

File: src/oop/app/src/ScrapperFactory.php (hunk 12 above)
Comment (AndriiBakayev) [reply src/oop/app/src/ScrapperFactory.php]: Done

File: src/oop/app/src/Models/Movie.php [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+    {
+        return $this->title;
+    }
+
+    /**
+     * @param string $title
+     */
+    public function setTitle(string $title): void
+    {
+            $this->title = $title;
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Models/Movie.php]: to much spaces

File: src/oop/app/src/ScrapperFactory.php [hunk 16]
```diff
@@ -2,28 +2,50 @@
...
+ * ScrapperFactory
+ */
 class ScrapperFactory
 {
-    /**
+    private Movie $movie;
+
+    // public function __construct(Movie $movieLeft, Movie $movieRight)
+    // {
+    //     $this->movieLeft = $movieLeft;
```
Comment (yderevianko-cartera) [inline src/oop/app/src/ScrapperFactory.php]: please remove commented lines

**Review by yderevianko-cartera — APPROVED** [review]

**Pull request merged**